assets/src/amp-validation/counts/index.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import { isPlainObject } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import domReady from '@wordpress/dom-ready';
+
+/**
+ * Internal dependencies
+ */
+import './style.css';
+
+/**
+ * Updates a menu item with its count.
+ *
+ * If the count is not a number or is `0`, the element that contains the count is instead removed (as it would be no longer relevant).
+ *
+ * @param {HTMLElement} itemEl Menu item element.
+ * @param {number} count Count to set.
+ */
+function updateMenuItem( itemEl, count ) {
+	count = Math.abs( count );
+
+	if ( isNaN( count ) || count === 0 ) {
+		itemEl.parentNode.parentNode.removeChild( itemEl.parentNode );
+	} else {
+		itemEl.classList.remove( 'loading' );
+		itemEl.textContent = count.toLocaleString();
+	}
+}
+
+/**
+ * Updates the 'Validated URLs' and 'Error Index' menu items with their respective unreviewed count.
+ *
+ * @param {any} counts Counts for menu items.
+ * @param {number} counts.validated_urls Unreviewed validated URLs count.
+ * @param {number} counts.errors Unreviewed validation errors count.
+ */
+function updateMenuItemCounts( counts ) {
+	if ( ! isPlainObject( counts ) ) {
+		// eslint-disable-next-line no-console
+		console.error( '[AMP Plugin] An error occurred while retrieving unreviewed validation counts. Received: ' + JSON.stringify( counts ) );
+		counts = {};
+	}
+
+	const { validated_urls: newValidatedUrlCount, errors: newErrorCount } = counts;
+
+	const errorCountEl = document.getElementById( 'new-error-index-count' );
+	const validatedUrlsCountEl = document.getElementById( 'new-validation-url-count' );
+
+	updateMenuItem( errorCountEl, newErrorCount );
+	updateMenuItem( validatedUrlsCountEl, newValidatedUrlCount );
+}
+
+domReady( async () => {
+	const counts = await apiFetch( { path: '/amp/v1/unreviewed-validation-counts' } );
```

Comment:
Another performance enhancement would be to check whether the `#toplevel_page_amp-options` element has the `wp-has-current-submenu` class meaning the submenu is expanded. If it does, then go ahead and fetch and display the counts. Otherwise, if it does not have this class (or if it has the `wp-not-current-submenu` class), then the menu is not expanded and it is not needed to do the fetch immediately. It can be deferred until the first time that the `opensub` class is added to this element, which is what happens when the user hovers over the collapsed menu item or tabs onto it. In this way, the REST API request will only be triggered with every admin page load if the user is looking at one of the AMP screens. If not, then the request will be deferred until the point that the user is actually going to see the menu.

Reply:
Actually, a nice way to implement both could be to wrap this logic in an `IntersectionObserver` callback, specifically targeting the parent element of the menu items. This should defer the loading of the data for 3 scenarios:

1. The user is not on an AMP screen so the menu items are not shown until the AMP menu is hovered/focus.
1. The browser window is too short to even show the initially-expanded AMP settings menu in the first viewport, meaning the fetch could be delayed until the user scrolls down.
1. The user has collapsed the entire admin menu in which case all of the submenus are hidden, even those which have the `wp-has-current-submenu` class.

Reply:
That worked perfectly (see https://github.com/ampproject/amp-wp/pull/5900/commits/70941a5d713af261113b6d4145fe28872145346d), thanks for the tip!